src/test/java/com/alkemy/ong/controller/NewsControllerTest.java

```diff
@@ -0,0 +1,362 @@
+                    .contentType(MediaType.APPLICATION_JSON)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .content(updatedContent);
+            mockMvc.perform(mockRequest)
+                    .andExpect(status().isOk());
+    }
+    @Test
+    @WithUserDetails(USER)
+    void TestPutNewsController_updateNews_Create_Role_USER() throws Exception {
+        NewsEntity newsEntity = createNewsEntity();
+        when(newsRepository.save(newsEntity)).thenReturn(newsEntity);
+        NewsEntity newsEntity2 = createNewsEntity2();
+        when(newsRepository.findById(newsEntity.getId())).thenReturn(java.util.Optional.of(newsEntity));
+        when(newsRepository.save(newsEntity2)).thenReturn(newsEntity2);
+        String updatedContent=objectWriter.writeValueAsString(newsEntity2);
+        MockHttpServletRequestBuilder mockRequest=MockMvcRequestBuilders.put("/news/123e4567-e89b-12d3-a456-426614174000")
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON)
+                .content(updatedContent);
+        mockMvc.perform(mockRequest)
+                .andExpect(status().isForbidden());
+    }
+    @Test
+    @WithUserDetails(ADMIN)
+    void TestDeleteNewsController_updateNews_Create_Role_ADMIN_Not_Found() throws Exception {
+        NewsDto newsDto= new NewsDto();
+        Throwable thrown = assertThrows(Exception.class, () -> newsController.update("1233",newsDto));
+        assertEquals("There are no news with the provided News id", thrown.getMessage());
+    }
+    @Test
+    @WithUserDetails(ADMIN)
+    void TestDeleteNewsController_eliminateNews_Create_Role_ADMIN() throws Exception {
+        MockHttpServletRequestBuilder mockRequest=MockMvcRequestBuilders.delete("/news/987")
+                    .contentType(MediaType.APPLICATION_JSON);
+        mockMvc.perform(mockRequest)
+                    .andExpect(status().isNoContent());
+    }
+    @Test
+    @WithUserDetails(ADMIN)
+    void TestDeleteNewsController_eliminateNews_Create_Role_ADMIN_Not_Found() throws Exception {
+        Throwable thrown = assertThrows(Exception.class, () -> newsController.delete("100"));
+        assertEquals("There are no news with the provided News id", thrown.getMessage());
+    }
+    @Test
+    @WithUserDetails(USER)
+    void TestGetNews_ValidatePaginationOk_Role_USER() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/news".concat("?page=")))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+    }
+    @Test
+    @WithUserDetails(ADMIN)
+    void TestGetNews_ValidatePaginationOk_Role_ADMIN() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/news".concat("?page=")))
+                .andExpect(MockMvcResultMatchers.status().isForbidden());
+    }
+
+
+    private NewsEntity createNewsEntity(){
```

Comment:
I suggest to implement a Mock class in package util

Reply:
Done.

Reply:
Looks good!